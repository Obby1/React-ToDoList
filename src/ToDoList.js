import React, { useState } from 'react';
import ToDoForm from './ToDoForm';
import ToDo from './ToDo';

const ToDoList = () =>{
    const [toDos, setToDos] = useState([])

    const addToDo = (todo) => {
        setToDos((toDos) => [...toDos, todo])
    }

    const removeToDo = (id) => {
        setToDos((toDos)=>toDos.filter((toDo) => toDo.id !== id))
    }

    
    const update = (id, text) => {
        setToDos(toDos =>
          toDos.map(todo =>
            todo.id === id ? { ...todo, text } : todo
          )
        );
      };

    return (
    <>
        <ToDoForm addToDo={addToDo} data-testid="new-todo-form"/>
        {toDos.map((toDo) => {
            return (
                <ToDo 
                    key={toDo.id}
                    id={toDo.id}
                    text={toDo.text}
                    removeToDo={removeToDo}
                    update={update} 
                    data-testid="todo" 
                />)})}
    </>
    )
}

export default ToDoList