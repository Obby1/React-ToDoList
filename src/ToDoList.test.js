import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ToDoList from './ToDoList';

it('renders ToDoList without crashing', () => {
    render(<ToDoList />);
  });
  
  it('matches ToDoList snapshot', () => {
    const { asFragment } = render(<ToDoList />);
    expect(asFragment()).toMatchSnapshot();
  });
  
  test('adds and removes a todo', () => {
    const { getByLabelText, getByText, getByTestId } = render(<ToDoList />);
    const textInput = getByLabelText('Text:');
    const addButton = getByText('Add To Do');
  
    fireEvent.change(textInput, { target: { value: 'Test ToDo' } });
    fireEvent.click(addButton);
  
    const todoElement = getByTestId('todo');
    expect(todoElement).toBeInTheDocument();
  
    const removeButton = getByText('X');
    fireEvent.click(removeButton);
    expect(todoElement).not.toBeInTheDocument();
  });
  
  test('edits a todo', async () => {
    const { getByLabelText, getByText, getByTestId } = render(<ToDoList />);
    const textInput = getByLabelText('Text:');
    const addButton = getByText('Add To Do');
  
    fireEvent.change(textInput, { target: { value: 'Test ToDo' } });
    fireEvent.click(addButton);
    const toDoText = getByText('Test ToDo')
    expect(toDoText).toBeInTheDocument();
  
    const editButton = getByText('Edit');
    fireEvent.click(editButton);
  
    // const updateButton = getByText('Update!');
    const updateButton = getByTestId('update-button');
    const updateInput = getByTestId('update-input');
    expect(updateInput).toBeInTheDocument();
    expect(updateButton).toBeInTheDocument();


    fireEvent.change(updateInput, { target: { value: 'Updated ToDo' } });
    fireEvent.click(updateButton);
    const updatedToDo = getByText('Updated ToDo');
    expect(updatedToDo).toBeInTheDocument();


  });


  