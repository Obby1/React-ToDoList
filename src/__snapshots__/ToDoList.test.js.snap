// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches ToDoList snapshot 1`] = `
<DocumentFragment>
  <form
    data-testid="new-todo-form"
  >
    <label
      for="text"
    >
      Text:
    </label>
    <input
      id="text"
      name="text"
      value=""
    />
    <button>
      Add To Do
    </button>
  </form>
</DocumentFragment>
`;
